### PR TITLE
Fix template sync

### LINKED ISSUE #208: Finish centralized template sync
- [x] make cruft use the release tag instead of `main`
- [x] install pre-commit in GH workflow to run`pre-commit autoupdate` successfully
- [x] fix fork support:
  - make the workflow clone into namespaced directories (e.g. `~/original-author/reponame`, `~/forking-user/reponame`)
  - make the branch contain the user name (e.g. `original-author-template-update-v3.7`)
- ~~[ ] make cruft runs and PR generations run in parallel (so waiting for GH happens in parallel)~~
- [ ] fix tests and merge PR

### PR DESCRIPTION
**Pull request by grst**
- [x] Wrap in try...except to keep updating other repos if one fails
- [x] dont make pr on repo if already exists for current version
- [x] exit 1, if at least one job has failed 
- [x] Execute on all repos (instead of just "infercnvpy")

Close #208

### CODE REVIEW

File: scripts/pyproject.toml [hunk 1]
```diff
@@ -23,6 +23,7 @@ dependencies = [
     "GitPython",
     "PyGitHub",
     "PyYAML",
+    "pre-commit"
```
Comment (flying-sheep) [inline scripts/pyproject.toml:26]: ???

**Review by flying-sheep — CHANGES_REQUESTED** [review]

File: scripts/pyproject.toml (hunk 1 above)
Comment (grst) [reply scripts/pyproject.toml:26]: no, it's required for the pre-commit autoupdate triggered by cruft when instantiating the template

Comment (flying-sheep) [reply scripts/pyproject.toml]: OK! I’ll add a comment to clarify that

File: scripts/src/scverse_template_scripts/cruft_prs.py [hunk 2]
```diff
@@ -211,7 +246,9 @@ def main(tag_name: str) -> None:
     release = get_template_release(con.gh, tag_name)
     repo_urls = get_repo_urls(con.gh)
     for repo_url in repo_urls:
-        make_pr(con, release, repo_url)
+        # TODO just use single-repo we control for testing
+        if repo_url.endswith("icbi-lab/infercnvpy"):
+            make_pr(con, release, repo_url)
```
Comment (flying-sheep) [inline scripts/src/scverse_template_scripts/cruft_prs.py:251]: don’t forget to fix this before merging

**Review by flying-sheep — CHANGES_REQUESTED** [review]

**Comment by grst** [comment]
@flying-sheep, do you have any clue why the test is failing? Locally all tests pass...

**Comment by grst** [comment]
The "real" sync run was successful 🎉. 

A couple of repos failed, but the script recovered gracefully. I set the respective repos to skip in https://github.com/scverse/ecosystem-packages/commit/10c060c7804d0435680f7f7cedaff767a422e219.

File: scripts/src/scverse_template_scripts/cruft_prs.py [hunk 3]
```diff
@@ -217,6 +222,11 @@ def make_pr(con: GitHubConnection, release: GHRelease, repo_url: str) -> None:
     # create fork, populate branch, do PR from it
     origin = con.gh.get_repo(repo_url.removeprefix("https://github.com/"))
     repo = get_fork(con, origin)
+
+    if old_pr := next((p for p in origin.get_pulls("open") if pr.matches_current_version(p)), None):
```
Comment (flying-sheep) [inline scripts/src/scverse_template_scripts/cruft_prs.py:226]:
```suggestion
    if any(True for p in origin.get_pulls("open") if pr.matches_current_version(p)):
```

**Review by flying-sheep — APPROVED** [review]

**Comment by flying-sheep** [comment]
awesome! Once the script tests are fixed properly, we can really call this done (well, actually expanding the tests wouldn’t hurt either)

**Comment by grst** [comment]
@flying-sheep, could you please take a look at the tests? I don't understand why they are failing and can't produce the failure locally.

**Pull request merged**